### DISCUSSION TITLE
Some questions about hostname clientHome apiURL

### DISCUSSION THREAD

**Discussion by Bunker56** [PhlexPlexico/G5API#127] [Help]
I can't configure the parameters of the `hostname`,`clientHome` and `apiURL` in any way, I went through all the options, but nothing works. For example, I have a server with an external ip 1.1.1.1, the domain name is host.test.com, LAN ip of this server- 2.2.2.2 (Internal LAN - 2.2.2.0/24). On this server - CS:GO, G5API and G5V. And that's where the fun begins. I can't figure out which IP and ports to use in order for the whole system to function normally? Help the noob, PLZ... :)

**Comment by PhlexPlexico** [comment]
Hiya,

So by the sounds of it - do you have a front end setup? I.e. G5V would be needed in order to set the clientHome. You can technically use just the API but you won't get any visual representation for any data that is stored.

I'll assume you have the front end setup and running, if not, please refer to the [wiki](https://github.com/PhlexPlexico/G5V/wiki/) on how to setup the front end, with reverse proxy and all. We'll assume that your front-end is hosted at `host.test.com`.

Next, there is a [wiki link](https://github.com/PhlexPlexico/G5API/wiki/Configuration) in how to configure this API. The `clientHome` config value is specifically used for redirecting authentication when signing in through steam. That shouldn't matter on anything to the game server. So, this value would simply be `"clientHome": "https://host.test.com",` provided it's hosted on port 80.

The hostname is what the API refers to itself as and what it tries to bind all calls to the api. In this case, it should be set to `https://host.test.com` as well. HTTPS is once again using an assumption you have a webserver setup as well (such as [NGINX](https://github.com/PhlexPlexico/G5V/wiki/Reverse-Proxy-Setup-with-NGINX) or [Caddy](https://github.com/PhlexPlexico/G5V/wiki/Webserver-Setup-With-Caddy)).

Next is the API URL, which is setup through as a reverse proxy (mentioned in the G5V installation). If you choose to not top have a reverse proxy setup, this can just be a direct link with your port to your API. Assuming G5API is hosted at `host.test.com:3301` you would set this value to `http://host.test.com:3301`.

And once again, this would be assuming that you have all these ports open/forwarded. Nothing will be able to communicate if these ports (such as 3301, or even port 80) are closed off. I believe both Wikis listed should be of enough help to get you up and running.

Thanks!

**Comment by Bunker56** [comment]
One more question. Are the same values used in Docker, or is there some nuance?

**Reply by PhlexPlexico** [reply]
All of it is listed in the readme for what you need to have defined as environment variables, this will copy a config inside the container with the provided values:

```
docker container run --name g5api \
-p 3301:3301 \
-v redisVol:/RedisFiles \
-e PORT="3301" \
-e HOSTNAME="" \
-e DBKEY="" \
-e STEAMAPIKEY="" \
-e SHAREDSECRET="" \
-e CLIENTHOME="" \
-e APIURL="" \
-e SQLUSER="" \
-e SQLPASSWORD="" \
-e DATABASE="" \
-e SQLHOST="" \
-e SQLPORT="" \
-e ADMINS="" \
-e SUPERADMINS="" \
-e REDISPASSWORD="" \
yourname\g5api:latest
```

Hostname would be the name of the container and Client home would be the address of the front end. I.e. if you were running the api on `api.host.com` and the front end was `webpanel.host.com` you would set those respectively.

**Comment by Bunker56** [comment]
I'm sorry, but I'm completely confused... :) Help the noob, because I really want to get away from eBot, but I can't figure it out. It seems like I'm doing everything as in the manual, but something does not work out in any way.

G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 15.464 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:00:23 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 3.152 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:00:28 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 3.215 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:01:20 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 4.792 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:01:22 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 2.816 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:01:26 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 2.309 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:01:29 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/vetoUpdate 500 1.935 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:01:29 +0000] "POST /match/6/vetoUpdate HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | Not a correct API Key.
G5API    | 0|G5API  | POST /match/6/map/0/start 500 5.063 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:03:45 +0000] "POST /match/6/map/0/start HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | Not a correct API Key.
G5API    | 0|G5API  | POST /match/6/map/0/update 500 5.174 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:04:35 +0000] "POST /match/6/map/0/update HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/map/0/player/76561199034716813/update 500 4.710 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:04:35 +0000] "POST /match/6/map/0/player/76561199034716813/update HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"
G5API    | 0|G5API  | POST /match/6/map/0/player/76561199034582085/update 500 4.531 ms - 36
G5API    | 0|G5API  | ::ffff:192.168.50.10 - - [13/Dec/2021:02:04:35 +0000] "POST /match/6/map/0/player/76561199034582085/update HTTP/1.1" 500 36 "-" "sm-ripext/1.3.1"

What's wrong?

**Reply by PhlexPlexico** [reply]
If you're getting POST errors, please make sure to look at the console log or the return value in the game console. If you're using Docker for this, the log file can be found inside the container under `/root/.pm2/logs/G5API-error.log` or if you're running it on a regular machine the log files are usually found in `~/.pm2/logs/G5API-error.log`. Alternatively, you can view the last few lines of logs by navigating to your G5API directory and running `yarn pm2 logs` to get the log outputs. The 500 error indicates that there is something wrong with the API and it's returning an error call.

**Comment by Bunker56** [comment]
This is my docker-compose.yml. 
Tell me where my mistake, plz...

version: '2.1'

networks:
  get5:
    external:
     name: bunker56

services:
  caddy:
    image: caddy:2-alpine
    container_name: caddy-reverse-proxy
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    volumes: 
      - ./Caddyfile:/etc/caddy/Caddyfile
    networks:
      - get5
    environment:
      - CADDY_URL=http://example.test.com
      - CADDY_API_ENDPOINT=api
      - CADDY_REVERSE_PROXY_PORT=3301
    networks:
      get5:
        ipv4_address: 192.168.50.10

  g5api:
    image: ghcr.io/phlexplexico/g5api:latest
    container_name: G5API
    volumes: 
      - redisVol:/RedisFiles
    expose:
      - 3301
    environment:
      - PORT=3301
      - HOSTNAME=example.test.com
      - DBKEY=e42d30c61d43305cb44f6h82f51df86f3
      - STEAMAPIKEY=3907D4B16CB6B17E5ED3745FC2380749
      - SHAREDSECRET=password
      - CLIENTHOME=http://example.test.com/
      - APIURL=http://example.test.com/api
      - SQLUSER=get5
      - SQLPASSWORD=password
      - SQLPORT=3306
      - DATABASE=get5
      - SQLHOST=192.168.50.2
      - ADMINS=76561188034716813
      - SUPERADMINS=76561188034716813
      - REDISPASSWORD=password
      - UPLOADDEMOS=true
    restart: unless-stopped
    networks:
      get5:
        ipv4_address: 192.168.50.11

  g5v:
    image: ghcr.io/phlexplexico/g5v:latest
    depends_on: 
      - g5api
      - caddy
    container_name: G5V-Front-End
    restart: always
    networks:
      get5:
        ipv4_address: 192.168.50.12


volumes:
  redisVol:
    external: false

**Reply by PhlexPlexico** [reply]
`- CADDY_URL=http://example.test.com` remove `http://`

`- HOSTNAME=example.test.com` should be `- HOSTNAME=https://example.test.com`

```
- CLIENTHOME=http://example.test.com/
- APIURL=http://example.test.com/api
``` 

Both should be`https://`

`- SQLHOST=192.168.50.2` not sure where your db is hosted, but the docker compose had the database built in to it.

```
networks:
get5:
ipv4_address: 192.168.50.12
```
Zero clue what this is doing here. Please use the docker compose file in the repo, and only change the values that are in the wiki. Also log files would be extremely helpful.

**Reply by PhlexPlexico** [reply]
Also you may want to remove or reset your steam API key :)
